library/Icingadb/Compat/UrlMigrator.php

```diff
@@ -84,9 +93,18 @@ public static function transformFilter(Filter\Rule $filter, $queryTransformer =
         }
 
         $rewritten = $transformer->rewrite($filter, $columns);
-        return $rewritten === false ? false : ($rewritten instanceof Filter\Rule ? $rewritten : $filter);
+        return $rewritten instanceof Filter\Rule ? $rewritten : $filter;
```

Comment:
This changes the implementation. `rewrite` may still return `false`.

Reply:
It doesn't matter what `rewrite()` is returning. If it isn't an instance of the `Filter\Rule`, the original filter will always be returned.